service-base/src/main/java/org/eclipse/hono/service/auth/device/DeviceCredentials.java

```diff
@@ -39,4 +44,17 @@
      * @return The tenant.
      */
     String getTenantId();
+
+    /**
+     * Gets additional properties that can be used to get credentials from the Credentials API.
+     * <p>
+     * This default implementation returns an empty JSON object.
+     *
+     * @return An empty JsonObject. An implementation can return a JSON object containing arbitrary properties. It is
+     *         intended to be directly usable to pass it into
+     *         {@link org.eclipse.hono.client.CredentialsClient#get(String, String, JsonObject, SpanContext)}.
+     */
+    default JsonObject getClientContext() {
+        return emptyClientContext;
```

Comment:
IMHO we should return a new instance here instead of a shared instance in order to prevent surprises when one client starts to add properties to the returned instance ...

Reply:
I was more worried about the surprise if it always returns a new instance. But you are right, this is not better. So, I'll change it.